src/DGtal/geometry/curves/Naive3DDSSComputer.h

```diff
@@ -0,0 +1,314 @@
+     * with the adding point if true.
+     * @return 'true' if yes, 'false' otherwise.
+     */
+    bool extendFront();
+
+
+    /**
+     * Tests whether the 3d DSS can be extended at the front.
+     *
+     * @return 'true' if yes, 'false' otherwise
+     */
+    bool isExtendableFront();
+
+    // ------------------------- Accessors ------------------------------
+
+    /**
+     * Computes the parameters
+     * (direction, intercept, thickness)
+     * of the DSS
+     * @param direction direction
+     * @param intercept intercept
+     * @param thickness thickness
+     */
+    void getParameters ( Vector3d& direction, PointD3d& intercept, PointD3d& thickness ) const;
+    void get2DSegmentsLength ( unsigned int & pLenXY, unsigned int & pLenXZ, unsigned int & pLenYZ ) const;
+
+    /**
+     * Checks the validity/consistency of the object.
+     * @return 'true' if the object is valid, 'false' otherwise.
+     */
+    bool isValid() const;
+
+    /**
+     *
+     * @return begin iterator of the 3d DSS range.
+     */
+    ConstIterator begin() const;
+    /**
+     * @return end iterator of the 3d DSS range.
+     */
+    ConstIterator end() const;
+
+    /**
+       @return a const-reference on the arithmetical DSS recognition
+       algorithm along the XY plane.
+    */
+    const ArithmeticalDSSComputer2d & arithmeticalDSS2dXY () const;
+
+    /**
+       @return a const-reference on the arithmetical DSS recognition
+       algorithm along the XZ plane.
+    */
+    const ArithmeticalDSSComputer2d & arithmeticalDSS2dXZ () const;
+
+    /**
+       @return a const-reference on the arithmetical DSS recognition
+       algorithm along the YZ plane.
+    */
+    const ArithmeticalDSSComputer2d & arithmeticalDSS2dYZ () const;
+
+    /**
+       @param i the axis orthogonal to the plane
+       @return a const-reference on the arithmetical DSS recognition
+       algorithm along the plane orthogonal to the \a i-th axis.
+    */
+    const ArithmeticalDSSComputer2d & arithmeticalDSS2d( Dimension i ) const;
+    
+    /**
+       @ param i the axis orthogonal to the plane
+       @return true if given 2D DSS along orthogonal axis is valid
+     */
+    bool validArithmeticalDSS2d ( Dimension i ) const;
+
+
+    // ------------------ Display ------------------------------------------
+  public:
+
+    /**
+     * @return the style name used for drawing this object.
+     */
+    std::string className() const;
+
+    /**
+     * Writes/Displays the object on an output stream.
+     * @param out the output stream where the object is written.
+     */
+    void selfDisplay ( std::ostream & out ) ;
+
+
+    // ------------------------- Protected Datas ------------------------------
+  protected:
+
+    /// projectors
+    Projector2d myProjXY, myProjXZ, myProjYZ;
+
+    /// 2d-arithmeticalDSS recognition algorithms
+    ArithmeticalDSSComputer2d myXYalgo;
+    ArithmeticalDSSComputer2d myXZalgo;
```

Comment:
Either do as above and puts all members in one line, or provide doc for all members.